File: 06-FEA-3D-Elasticity/Matlab-Code/PrintPostProcessingResults.m
function PrintPostProcessingResults(StressAtIntegrationPoints, ...
    StrainAtIntegrationPoints, StressAtCentroid, StrainAtCentroid, ...
    StressAtNodes, StrainAtNodes, TpAtIntegrationPoints, ...
    VMAtIntegrationPoints, TpAtCentroid, TpAtNodes, VMAtNodes)

fprintf('STRESS AT INTEGRATION POINTS\n')
disp(StressAtIntegrationPoints);
fout = fopen('StressAtIntegrationPoints.dat','w');
for i = 1:size(StressAtIntegrationPoints,1)
    fprintf(fout, '%12.4f ',StressAtIntegrationPoints(i,:));
    fprintf(fout, '\n');
end

fprintf('STRAIN AT INTEGRATION POINTS\n')
disp(StrainAtIntegrationPoints);
fout = fopen('StrainAtIntegrationPoints.dat','w');
for i = 1:size(StrainAtIntegrationPoints,1)
    fprintf(fout, '%12.4f ',StrainAtIntegrationPoints(i,:));
    fprintf(fout, '\n');
end

fprintf('PRINCIPAL STRESSES AT INTEGRATION POINTS\n')
disp(TpAtIntegrationPoints);
fout = fopen('PrincipalStressesAtIntegrationPoints.dat','w');
for i = 1:size(TpAtIntegrationPoints,1)
    fprintf(fout, '%12.4f ',TpAtIntegrationPoints(i,:));
    fprintf(fout, '\n');
end

fprintf('VON MISES STRESSES AT INTEGRATION POINTS\n')
disp(VMAtIntegrationPoints);
fout = fopen('VonMisesAtIntegrationPoints.dat','w');
for i = 1:size(VMAtIntegrationPoints,1)
    fprintf(fout, '%12.4f ',VMAtIntegrationPoints(i,1));
    fprintf(fout, '\n');
end

fprintf('STRESS AT CENTROID\n')
disp(StressAtCentroid);
fout = fopen('StressAtCentroid.dat','w');
for i = 1:size(StressAtCentroid,1)
    fprintf(fout, '%12.4f ',StressAtCentroid(i,:));
    fprintf(fout, '\n');
end

fprintf('STRAIN AT CENTROID\n')
disp(StrainAtCentroid);
fout = fopen('StrainAtCentroid.dat','w');
for i = 1:size(StrainAtCentroid,1)
    fprintf(fout, '%12.4f ',StrainAtCentroid(i,:));
    fprintf(fout, '\n');
end

fprintf('PRINCIPAL STRESSES AT CENTROID\n')
disp(TpAtCentroid);
fout = fopen('PrincipalStressesAtCentroid.dat','w');
for i = 1:size(TpAtCentroid,1)
    fprintf(fout, '%12.4f ',TpAtCentroid(i,:));
    fprintf(fout, '\n');
end

fprintf('STRESS AT NODES\n')
disp(StressAtNodes);
fout = fopen('StressAtNodes.dat','w');
for i = 1:size(StressAtNodes,1)
    fprintf(fout, '%12.4f ',StressAtNodes(i,:));
    fprintf(fout, '\n');
end

fprintf('STRAIN AT NODES\n')
disp(StrainAtNodes);
fout = fopen('StrainAtNodes.dat','w');
for i = 1:size(StrainAtNodes,1)
    fprintf(fout, '%12.4f ',StrainAtNodes(i,:));
    fprintf(fout, '\n');
end

fprintf('PRINCIPAL STRESSES AT NODES\n')
disp(TpAtNodes);
fout = fopen('PrincipalStressesAtNodes.dat','w');
for i = 1:size(TpAtNodes,1)
    fprintf(fout, '%12.4f ',TpAtNodes(i,:));
    fprintf(fout, '\n');
end


fprintf('VON MISES STRESSES AT NODES\n')
disp(VMAtNodes);
fout = fopen('VonMisesStressesAtNodes.dat','w');
for i = 1:size(VMAtNodes,1)
    fprintf(fout, '%12.4f ',VMAtNodes(i));
    fprintf(fout, '\n');
end

end

File: 06-FEA-3D-Elasticity/C-Code/PrintPostProcessingResults.cpp
#include <MainFile.h>
using namespace std;

void PrintPostProcessingResults(arma::mat& StrainAtIntegrationPoints, arma::mat& StressAtIntegrationPoints,
        arma::mat& TpAtIntegrationPoints, arma::mat& VMAtIntegrationPoints,
        arma::mat& StrainAtCentroid, arma::mat& StressAtCentroid, arma::mat& TpAtCentroid,
        arma::mat& StrainAtNodes, arma::mat& StressAtNodes, arma::mat& TpAtNodes, arma::mat&  VMAtNodes)
{
  ofstream fout;
  
    cout<<"STRESS AT INTEGRATION POINTS\n";
  cout << StressAtIntegrationPoints;
  fout.open ("StressAtIntegrationPoints.dat");
  fout << StressAtIntegrationPoints;
  fout.close();
  
  cout<<"STRAIN AT INTEGRATION POINTS\n";
  cout << StrainAtIntegrationPoints;
  fout.open ("StrainAtIntegrationPoints.dat");
  fout << StrainAtIntegrationPoints;
  fout.close();
  
  cout<<"PRINCIPAL STRESSES AT INTEGRATION POINTS\n";
  cout << TpAtIntegrationPoints;
  fout.open ("PrincipalStressesAtIntegrationPoints.dat");
  fout << TpAtIntegrationPoints;
  fout.close();
  
  cout<<"VOM MISES STRESSES AT INTEGRATION POINTS\n";
  cout << VMAtIntegrationPoints;
  fout.open ("VonMisesAtIntegrationPoints.dat");
  fout << VMAtIntegrationPoints;
  fout.close();

  cout<<"STRESS AT CENTROID\n";
  cout << StressAtCentroid;
  fout.open ("StressAtCentroid.dat");
  fout << StressAtCentroid;
  fout.close();
  
  cout<<"STRAIN AT CENTROID\n";
  cout << StrainAtCentroid;
  fout.open ("StrainAtCentroid.dat");
  fout << StrainAtCentroid;
  fout.close();
  
  cout<<"PRINCIPAL STRESSES AT CENTROID\n";
  cout << TpAtCentroid;
  fout.open ("PrincipalStressesAtCentroid.dat");
  fout << TpAtCentroid;
  fout.close();
  

  cout<<"STRESSES AT NODES\n";
  cout << StressAtNodes;
  fout.open ("StressAtNodes.dat");
  fout << StressAtNodes;
  fout.close();
  
  cout<<"STRAIN AT NODES\n";
  cout << StrainAtNodes;
  fout.open ("StrainAtNodes.dat");
  fout << StrainAtNodes;
  fout.close();
 
  cout<<"PRINCIPAL STRESSES AT NODES\n";
  cout << TpAtNodes;
  fout.open ("PrincipalStressesAtNodes.dat");
  fout << TpAtNodes;
  fout.close();
  
  cout<<"VOM MISES STRESSES AT NODES\n";
  cout << VMAtNodes;
  fout.open ("VonMisesStressesAtNodes.dat");
  fout << VMAtNodes;
  fout.close();
}